thunder/tests/test_dynamo.py

```diff
@@ -1123,3 +1124,30 @@ def check(file_name, cmd):
 
     for file in py_files:
         check(tmp_path / file, cmd)
+
+
+def test_leak_on_unsupported_thunder_operator():
+    # This test is to check the fix for a previous leak
+    # which was caused by holding onto the
+    # exception object in split_reason.
+
+    def unsupported_op_fn(w1) -> torch.Tensor:
+        topk_ids = torch.tensor([[0, 1]])
+        # This indexing is not supported by thunder and get's passed to inductor.
+        w13_weights = w1[topk_ids]
+        return w13_weights + 1
+
+    def call_thunderfx_on_leaking_fn():
+        w1 = torch.randn(16, 16, 32, dtype=torch.bfloat16)
+        fn = thunderfx(unsupported_op_fn)
+        fn(w1)
+
+        # There should be two thunder traces because of the split caused by indexing.
+        # In future when thunder supports this indexing, the test will fail here
+        # as we won't be checking for the leak in case of unsupported operation.
+        assert len(fn.last_traces) == 2
+
+        return weakref.ref(w1)
+
+    t_weak_ref = call_thunderfx_on_leaking_fn()
```

Comment:
This seems OK since it's passing CI, although in the future I'm not sure if we can rely on CPython to delete `w1` without calling https://docs.python.org/3/library/gc.html#gc.collect explicitly

Reply:
We may want to do something like call https://docs.python.org/3/library/gc.html#gc.get_referents on the reports